File: python/linked-lists/0328-odd-even-linked-list.py
from typing import ListNode


def oddEvenList(head: ListNode) -> ListNode:
    odd = head
    if head:
        even = head.next
        
        cur_odd = odd
        cur_even = even
        while cur_odd.next and cur_even.next:
            cur_odd.next = cur_odd.next.next
            cur_odd = cur_odd.next
            cur_even.next = cur_even.next.next
            cur_even = cur_even.next
            
        cur_odd.next = even
        
    return odd

File: js/linked-lists/0328-odd-even-linked-list.js
var oddEvenList = function(head) {
    if (head === null) return head;
    
    var odd = head;
    var even = head.next;
    var evenHead = head.next;
    
    while (odd.next !== null && even.next !== null) {
        odd.next = even.next;
        even.next = odd.next.next;
        odd = odd.next;
        even = even.next;
    }
    
    odd.next = evenHead;
    
    return head;
};